src/core/Akka/Actor/ActorCell.cs

```diff
@@ -518,9 +518,12 @@ public static IActorRef GetCurrentSenderOrNoSender()
 
         private Envelope SerializeAndDeserialize(Envelope envelope)
         {
-            DeadLetter deadLetter;
-            var unwrapped = (deadLetter = envelope.Message as DeadLetter) != null ? deadLetter.Message : envelope.Message;
+            var unwrapped = envelope.Message;
+            
+            if(envelope.Message is IWrappedMessage wrapped)
```

Comment:
for all `IWrappedMessage`s: if the underlying message doesn't want to be serialized (typically because it can't be), don't try it. Just return the message without performing any additional work.